spec/forms/admin_reservation_form_spec.rb

```diff
@@ -0,0 +1,108 @@
+require "rails_helper"
+
+RSpec.describe AdminReservationForm do
+  include DateHelper
+
+  let(:instrument) { create(:setup_instrument) }
+  let(:start_date) { Time.zone.parse("2017-10-17 12:00:00") }
+  let(:admin_reservation) { build(:admin_reservation, product: instrument, reserve_start_at: start_date, duration: 1.hour) }
+  let!(:form) { AdminReservationForm.new(admin_reservation) }
+
+  describe "validations" do
+    describe "cannot_exceed_max_end_date" do
+      it "invalid past max end date" do
+        form.assign_attributes(repeats: "1", repeat_frequency: "weekly", repeat_end_date: format_usa_date(start_date + 14.weeks))
+        expect(form.valid?).to eq false
```

Comment:
In addition to this validity check, we should also check `expect(form.errors).to be_added(:repeat_end_date, :too_far_in_future)`. That way we see it's invalid for the right reason.

And, it's a matter of taste, but I lean towards `expect(form).not_to be_valid`.